### PR TITLE
Improve Screen Reader Accessibility + Add TypeScript helper Function `invalidSwitchCase`

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### Link to issue
https://reload.atlassian.net/browse/DDFSOEG-435
https://reload.atlassian.net/browse/DDFSOEG-439
https://reload.atlassian.net/browse/DDFSOEG-443

#### Description
This pull request includes several improvements to enhance accessibility for users who rely on screen readers, as well as adding a TypeScript helper function named `invalidSwitchCase`. 

The primary changes are as follows:

- Added two translation functions, `getFacetFieldTranslation` and `getReservationModalTypeTranslation`, which utilize the `invalidSwitchCase` helper function to ensure all cases are covered in a switch function.

- Updated various elements with appropriate translations for their 'aria-label' attributes and other relevant places.

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

File: src/components/facet-line/FacetLineFilters.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ const FacetLineFilters: React.FunctionComponent<FacetLineFiltersProps> = ({
                   value: ""
                 }}
                 options={formatValuesToDropdown(name, values)}
-                ariaLabel={name}
+                ariaLabel={t(`facet${upperFirst(name)}Text`)}
```
Comment (kasperg) [inline src/components/facet-line/FacetLineFilters.tsx:63]: This creates a tricky situation for managing translations as the code does not expose any information about what strings we can expect. This also shows by the fact that no facet translation controls are added in Storybook.

As I see it the possible facet names are limited by [the `FacetField` enumeration](https://github.com/reload/dpl-react/blob/main/src/core/dbc-gateway/generated/graphql.tsx#L306-L306). 


If you want to take a stab at it consider creating a function for generating translations for facet names and use TypeScript to ensure that all values are covered and translations are explicit but I do not think it is necessary to move this PR along.

File: src/components/reservation/forms/ModalReservationFormSelect.tsx [hunk 2]
```diff
@@ -22,9 +23,9 @@ const modalProps = (
 ) => ({
   modalId: modalReservationFormId(type),
   screenReaderModalDescriptionText: t(
-    `screenReaderModalDescription${type.toUpperCase()}Text`
+    `screenReaderModalDescription${upperFirst(type)}Text`
```
Comment (kasperg) [inline src/components/reservation/forms/ModalReservationFormSelect.tsx:26]: See comment about regarding dynamically generated translation strings. This case and the rest of this PR is not so bad though.

File: src/apps/material/material.entry.tsx [hunk 3]
```diff
@@ -97,16 +101,16 @@ interface MaterialEntryTextProps {
...
-  orderDigitalCopyFeedbackOkText: string;
   orderDigitalCopyFeedbackErrorAgencyNotSubscribedText: string;
   orderDigitalCopyFeedbackErrorInvalidPickupBranchText: string;
   orderDigitalCopyFeedbackErrorMissingClientConfigurationText: string;
   orderDigitalCopyFeedbackErrorPidNotReservableText: string;
   orderDigitalCopyFeedbackErrorUnauthenticatedUserText: string;
+  orderDigitalCopyFeedbackOkText: string;
   orderDigitalCopyFeedbackTitleText: string;
+  orderDigitalCopyModalCloseModalAriaLabelText: string;
+  orderDigitalCopyModalScreenReaderModalDescriptionText: string;
```
Comment (kasperg) [inline src/apps/material/material.entry.tsx:113]: :+1: for cleanup.

**Review by kasperg — APPROVED** [review]
:+1: LGTM

File: src/components/facet-browser/helper.ts [hunk 4]
```diff
@@ -100,4 +100,41 @@ export function getAllFilterPathsAsString(filterObject: {
   return allFilterPathsAsString;
 }
 
+const handleFacetFieldTranslationError = (name: string): never => {
+  throw new Error(`Didn't expect "${name}" in getFacetFieldTranslation`);
+};
+
+export const getFacetFieldTranslation = (name: string): string => {
```
Comment (kasperg) [inline src/components/facet-browser/helper.ts:107]: As far as I can tell this addition needs some more work and should not be merged as is.

**Review by kasperg — CHANGES_REQUESTED** [review]

**Review by sinejespersen — APPROVED** [review]

**Review by kasperg — APPROVED** [review]
:+1: LGTM

**Pull request merged**